### PR TITLE
Document non-standard string operators

### PR DESCRIPTION
**Pull request by 4tXJ7f**
This adds documentation for `str.indexof_re`, `str.update`, `str.rev`,
`str.to_lower`, and `str.to_upper`. Note that it moves the documentation
of strings to the "non-standard or extended theories".

### CODE REVIEW

File: docs/theories/strings.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+<https://smtlib.cs.uiowa.edu/theories-UnicodeStrings.shtml>`_. It additionally
+supports some non-standard operators that are described below.
+
+Semantics
+^^^^^^^^^
+
+.. code-block::
+
+   * (str.indexof_re String RegLan Int Int)
+
```
Comment (barrettcw) [inline docs/theories/strings.rst:14]: Why are there 3 arguments?  It looks like the third argument is not used in the definition.

**Review by barrettcw — CHANGES_REQUESTED** [review]

File: docs/theories/theories.rst [hunk 2]
```diff
@@ -21,7 +21,6 @@ Standardized theories
    Theory of integer arithmetic <https://smtlib.cs.uiowa.edu/theories-Ints.shtml>
    Theory of real arithmetic <https://smtlib.cs.uiowa.edu/theories-Reals.shtml>
    Theory of mixed-integer arithmetic <https://smtlib.cs.uiowa.edu/theories-Reals_Ints.shtml>
-   Theory of strings <https://smtlib.cs.uiowa.edu/theories-UnicodeStrings.shtml>
```
Comment (ajreynol) [inline docs/theories/theories.rst:24]: I think it makes sense to still list this here.  The other extensions of theories (e.g. transcedentals extend arithmetic), we keep the link to the standard theory here.

**Review by ajreynol — CHANGES_REQUESTED** [review]
One minor request else LGTM

File: docs/theories/strings.rst (hunk 1 above)
Comment (4tXJ7f) [reply docs/theories/strings.rst:14]: Good catch. I have updated the definition.

File: docs/theories/theories.rst (hunk 2 above)
Comment (4tXJ7f) [reply docs/theories/theories.rst]: Done.

**Review by ajreynol — APPROVED** [review]

**Review by barrettcw — APPROVED** [review]
LGTM

**Pull request merged**